### ISSUE TITLE
[BUG] Upgrade failed with "no route to host" error while retrieving repo info

### ISSUE THREAD

**Issue by starbops** [harvester/harvester#3550]
**Describe the bug**

This one is similar to #3444 but has an uncaught networking error: "no route to host."

<img width="762" alt="Screen Shot 2023-02-24 at 18 18 32" src="https://user-images.githubusercontent.com/1827717/221155001-f7d4141f-fb5d-435d-b892-8110076d8782.png">

The log lines from the `harvester` Pod:

```
...
time="2023-02-24T07:58:41Z" level=info msg="Starting upgrade repo VM"
time="2023-02-24T07:58:41Z" level=info msg="handle upgrade harvester-system/hvst-upgrade-vh6bs with labels map[harvesterhci.io/latestUpgrade:true harvesterhci.io/upgradeState:PreparingRepo]"
W0224 07:58:56.281689       7 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
time="2023-02-24T07:59:37Z" level=info msg="handle upgrade harvester-system/hvst-upgrade-vh6bs with labels map[harvesterhci.io/latestUpgrade:true harvesterhci.io/upgradeState:RepoPrepared]"
time="2023-02-24T07:59:44Z" level=warning msg="Repo info retrieval failed with: Get \"http://upgrade-repo-hvst-upgrade-vh6bs.harvester-system/harvester-iso/harvester-release.yaml\": dial tcp 10.53.230.35:80: connect: no route to host"
time="2023-02-24T07:59:44Z" level=info msg="handle upgrade harvester-system/hvst-upgrade-vh6bs with labels map[harvesterhci.io/latestUpgrade:true harvesterhci.io/upgradeState:Failed]"
time="2023-02-24T07:59:44Z" level=info msg="Add OwnerReference upgrade-repo-hvst-upgrade-vh6bs-disk-0 to upgrade repo image harvester-system/image-stlwr"
time="2023-02-24T07:59:44Z" level=info msg="Delete upgrade repo VM harvester-system/upgrade-repo-hvst-upgrade-vh6bs"
...
```

**To Reproduce**
1. Deploy a Harvester cluster (single node is okay) with master-head build
2. Upgrade with the same version ISO image
3. After the repo VM is up, chances are the upgrade will fail (not always reproducible)

**Expected behavior**

The upgrade should succeed.

**Support bundle**

**Environment**
 - Harvester ISO version: master
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by TachunLin** [comment]
Verified **not fixed** on `master-e341a72d-head` (03/06). Move the issue to implement.

### Result
While upgrading from master to master release, it will prompt the following error at downloading upgrade image stage
  ```
  Get "http://upgrade-repo-hvst-upgrade-p47zw.harvester-system/harvester-iso/harvester-release.yaml": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
  ```
  ![image](https://user-images.githubusercontent.com/29251855/223047269-ba7695a5-a048-468d-8472-76c7c0de2a20.png)

### Test Information
* Test Environment: 1 node Harvester on local kvm machine
* Harvester version: `master-e341a72d-head` (03/06)

### Verify Steps

1. Deploy a Harvester cluster (single node is okay) with the fix
2. Upgrade with the same ISO image that used to install the cluster
3. The upgrade should succeed

### Additional Context
* support bundle: 
[supportbundle_56a7159c-906e-49f5-a898-6744e2088178_2023-03-06T07-44-16Z.zip](https://github.com/harvester/harvester/files/10895508/supportbundle_56a7159c-906e-49f5-a898-6744e2088178_2023-03-06T07-44-16Z.zip)

**Comment by TachunLin** [comment]
Verified **fixed** on `master-2f37e2be-head` (03/07). Close this issue.

### Result
After 15 time upgrade trial test, we did not encounter the error of `not route to host` or `client timeout exeed`
![image](https://user-images.githubusercontent.com/29251855/223390291-b63deb12-c7d1-4079-a718-67629bb64b1b.png)


### Test Information
* Test Environment: 1 node Harvester on local kvm machine
* Harvester version: `master-2f37e2be-head` (03/07)

### Verify Steps
1. Deploy a Harvester cluster (single node is okay) with the fix
2. Upgrade with the same ISO image that used to install the cluster
3. The upgrade should succeed to the upgrading node stage
